json-schema-check/validate_json_schemas.py

```diff
@@ -0,0 +1,48 @@
+import os
+import subprocess
+import glob
+import colorama
+
+colorama.init(autoreset=True, strip=False)
+
+def main():
+    validation_failed = False
+
+    for schema_path in glob.glob('./**/schema/*.schema.json', recursive=True):
+        schema_file = os.path.relpath(schema_path, './')
+        base_name = os.path.basename(schema_file).replace('.schema.json', '')
+        config_dir = os.path.dirname(schema_file).replace('schema', 'config')
+
+        config_files = glob.glob(f'{config_dir}/{base_name}*.param.yaml')
+        if not config_files:
+            print(colorama.Fore.YELLOW + f'No configuration files found for schema {schema_file}.')
+            continue
+
+        for config_file in config_files:
+            print(
+                colorama.Style.BRIGHT + '🔍 Validating: ' +
+                colorama.Style.RESET_ALL +
+                colorama.Fore.CYAN + f'{config_file} ' +
+                colorama.Style.RESET_ALL + '🆚 ' +
+                colorama.Fore.BLUE + f'{schema_file}' +
+                colorama.Style.RESET_ALL,
+                end=' '
+            )
+            result = subprocess.run(['check-jsonschema', '--schemafile', schema_file, config_file], capture_output=True)
+            if result.returncode != 0:
+                print(colorama.Fore.RED + '❌ Failed')
+                for line in result.stdout.decode('utf-8').split('\n'):
+                    if line:
+                        print(colorama.Fore.RED + line)
```

Comment:
Maybe we could indent those a little? If it makes sense, then probably the "no configuration files found" should be indented by the same amount

Reply:
Change: https://github.com/autowarefoundation/autoware-github-actions/pull/285/commits/2aec9b2c8a1b9f2bd42b40b948b1b7c88c1999d0

Example run: https://github.com/autowarefoundation/autoware.universe/actions/runs/7981506848/job/21813925896

![Screenshot from 2024-02-21 15-08-54](https://github.com/autowarefoundation/autoware-github-actions/assets/10751153/f6996099-2b5d-4dfe-b6cc-62e52cdbbb6f)

This is what you've meant right?

Reply:
yes, perfect